instrumentation/http-url-connection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlConnectionSingletons.java

```diff
@@ -34,11 +36,39 @@ public final class HttpUrlConnectionSingletons {
             .addAttributesExtractor(PeerServiceAttributesExtractor.create(netAttributesGetter))
             .addOperationMetrics(HttpClientMetrics.get())
             .newClientInstrumenter(RequestPropertySetter.INSTANCE);
+
+    SunHttpUrlHttpAttributesGetter sunHttpUrlHttpAttributesGetter =
+        new SunHttpUrlHttpAttributesGetter();
+    SUN_INSTRUMENTER =
+        Instrumenter.<HttpURLConnection, Integer>builder(
+                GlobalOpenTelemetry.get(),
+                "io.opentelemetry.http-url-connection-sun",
+                HttpSpanNameExtractor.create(sunHttpUrlHttpAttributesGetter))
+            .setSpanStatusExtractor(HttpSpanStatusExtractor.create(sunHttpUrlHttpAttributesGetter))
+            .addAttributesExtractor(
+                HttpClientAttributesExtractor.create(sunHttpUrlHttpAttributesGetter))
+            .addAttributesExtractor(NetClientAttributesExtractor.create(netAttributesGetter))
+            .addAttributesExtractor(PeerServiceAttributesExtractor.create(netAttributesGetter))
+            .addOperationMetrics(HttpClientMetrics.get())
+            .newClientInstrumenter(RequestPropertySetter.INSTANCE);
   }
 
-  public static Instrumenter<HttpURLConnection, Integer> instrumenter() {
+  private HttpUrlConnectionSingletons() {}
+
+  public static Instrumenter<HttpURLConnection, Integer> instrumenter(
+      Class<? extends HttpURLConnection> connectionClass, String methodName) {
+
+    if (isGetOutputStreamMethodOfSunConnection(connectionClass, methodName)) {
+      return SUN_INSTRUMENTER;
+    }
```

Comment:
Is this issue really specific to the `com.sun` implementation? Can we simplify the condition just to the method name?